### ISSUE TITLE
[Feature request] Tab/Tree Context menu hotkey for TST

### ISSUE THREAD

**Issue by SHHSSH** [piroor/treestyletab#3450]
Hi Piroor,

As one would elsewhere in an operating system for files etc. with "Appplication" keys on some keyboards, is there at all the possibility for a hotkey in the options for the user to open the context menu for the current, or, selected tab?
https://ibb.co/yhvtv2n
It significantly increases efficiency for the user, whereby saving them the physical movement of the mouse cursor to the sidebar and would allow users to then henceforth press the corresponding access key.

***EDIT*** This is probably out of scope as you'd say due to most helper addons and TST itself offering a myriad of hotkey customisation in their options. If so, I apologise and close this issue, I just thought it could be very useful universally in FF, but if you're not interested in it all, no worries :)

Thanks in advance, as always for any consideration at all.

**Comment by piroor** [comment]
Which is an application your screenshot is taken about? My keyboard also has the application key (I think you meant the key with an menu-like icon), but Firefox looks to handle pressed app key as pressed on the main content area. After I click a sidebar contents the event for pressed app key is notified to the sidebar contents, so the behavior you described looks to be possible to implement, but you still need to click once on the sidebar, thus it looks inconvenient...

Anyway I've introduced a change to treat pressing of the application key as "context menu on the active tab" with the commit 6a903e4.

**Comment by SHHSSH** [comment]
Oh, yes, it's referred to as the Menu key. It has the functionality of Shift+F10 in windows for context menu I think?
I don't really know if it's really relevant to this though, I just used it as a reference for its context menu opening capabilities.

**Comment by SHHSSH** [comment]
Oh, with the likelihood of having to click on the sidebar once for focus of sorts, perhaps I will work around that with another macro for its focus in conjunction with the opening of the menu(if possible). 

Thanks for your consideration, I hope to have intrigued you somewhat.

**Comment by SHHSSH** [comment]
@piroor 
Is there any possible way to have this work without having TST sidebar "clicked/focused" as it's just not pratical to click onto TST sidebar then use the application key (menu-like icon key), at that point you masewell just right click the tab.

- I am inquring to see if there is any potential option for sending this key's function to TST perhaps via an optional hotkey?
- I understand the functionality is working if TST sidebar is focused, however, this is not any form of increased productivity.
- I also understand that it is not feasible to allow for a universal command of "application key/menu-like icon key) to be able to be used as that would conflict with general universal webpage usage of the key.

_I believe this would be immensely utilised by the community by providing an **even more** seamless productive flow allowing them to essentially "right click" on the current tab._

Thank in advance for any consideration as per usual.

> #### No need to read; 
> ##### Stupid needless information about my current workaround
> `At the moment I was trying to do workarounds with Alt-Tabbing to Show TST in New Window via multiple macros that were being scripted, then sending the application key, however, this is not a viable solution as when using multiple windows, they conflict with one another when trying to easily access multiple TSTs in New Window. My workaround is just not viable when using multiple windows, besides the fact it's just stupid anyway, having to press like 3+ keys for something that should be easier.`

**Comment by piroor** [comment]
Sadly it is impossible for now.

* Firefox does not allow WebExtensions API based addons to provide ability to open native context menu by script.
* Instead we can open the "fake" context menu simulated by TST itself (you'll see it by Ctrl-right-click on a tab in the sidebar.)
* However you still cannot use cursor keys to move focus in the menu, while the sidebar contents is not focused. We need to define new keyboard shortcut slots and assign custom keyboard shortcuts like "Move Up Focus in the Fake Context Menu" to "Alt-Shift-Up" or others.

Thus, I think it won't be enough useful. Something utility application for gamers providing ability to send arbitrary message to applications may be better solution for this situation.

**Comment by SHHSSH** [comment]
>     * Instead we can open the "fake" context menu simulated by TST itself (you'll see it by Ctrl-right-click on a tab in the sidebar.)

Yep, I am aware of this actually with the Ctrl-right-click, can this be invoked by a hotkey at all though?
Perhaps that's the solution? Although it does not provide the user with third-party addon context menu options, TST's context menu options are likely sufficient for many and me too. As with native Firefox context menu I don't even use many additional third party context menu options (only one "disable tab detach" but it's not that often used.
![capture_002_26062024_055010](https://github.com/piroor/treestyletab/assets/21285581/59d135d9-2ca7-47d5-8b9f-0af56b8dcb6b)

So would a hotkey implementation option be a reasonable approach here if that's at all possible to invoke TST's context menu?

If not, all good mate, cheers for the clarification.

**Comment by irvinm** [comment]
@piroor can we close this?

**Comment by piroor** [comment]
After some trials I've realized that you can move focus to the sidebar area only with the keyboard with the following key sequence:

1. Hit the F6 key to move focus to the address bar.
2. Hit the F6 key again to move focus to the sidebar header area.
3. Hit the Tab key to move focus to the sidebar contents.

If your OS supports opening of the native context menu via your keyboard, you can open the menu without any new custom shortcut implemented by TST side. For example, if you use a Windows PC you can open the context menu by:

* Shift-F10
* the context menu key (it may be placed next to the right "Windows" key, if you use a generic keyboard for Windows PC)

So I've introduced a change fe425e9 to treat the active tab as the context tab when the context menu is opened via keyboard operations. I hope the change helps you.
